homeassistant/components/smarttub/sensor.py

```diff
@@ -36,17 +44,69 @@ async def async_setup_entry(hass, entry, async_add_entities):
 class SmartTubSensor(SmartTubEntity):
     """Generic and base class for SmartTub sensors."""
 
-    def __init__(self, coordinator, spa, sensor_name, spa_status_key):
+    def __init__(self, coordinator, spa, sensor_name, attr_name):
         """Initialize the entity."""
         super().__init__(coordinator, spa, sensor_name)
-        self._spa_status_key = spa_status_key
+        self._attr_name = attr_name
 
     @property
     def _state(self):
         """Retrieve the underlying state from the spa."""
-        return self.get_spa_status(self._spa_status_key)
+        return getattr(self.spa_status, self._attr_name)
 
     @property
     def state(self) -> str:
         """Return the current state of the sensor."""
+        if isinstance(self._state, Enum):
+            return self._state.name.lower()
         return self._state.lower()
+
+
+class SmartTubPrimaryFiltrationCycle(SmartTubSensor):
+    """The primary filtration cycle."""
+
+    def __init__(self, coordinator, spa):
+        """Initialize the entity."""
+        super().__init__(
+            coordinator, spa, "primary filtration cycle", "primary_filtration"
+        )
+
+    @property
+    def state(self) -> str:
+        """Return the current state of the sensor."""
+        return self._state.status.name.lower()
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes."""
+        state = self._state
+        return {
+            ATTR_DURATION: state.duration,
```

Comment:
What does the duration attribute measurement return?

Reply:
It is an integer number of hours representing the desired length of the cycle. I'll add a comment.

Reply:
Addressed in https://github.com/home-assistant/core/pull/47192